### PR TITLE
Compare Only Final Submissions When Report Downloading

### PR DESCRIPTION
**Pull request by awaisdar001**
#### What are the relevant tickets?
[TNL-7511](https://openedx.atlassian.net/browse/TNL-7511)

#### What's this PR do?
There is a bug in getting students' submissions for comparison with the report generated for submissions.  There are two types of submissions, Finalised (File Uploaded and Submitted) and non-finalized (File Uploaded Only). When you click **Download All Submissions**,
Screen Shot 2021-02-09 at 7 03 26 PM

The system fetches _current_ student submissions and compares them when the submissions created in the last report. If the count differs, the system updates/replaces the generated report. There was a bug in this comparison. The report is generated for both types of submissions while the current/recent submission count is fetched only for **Finalized** submissions due to which the count is never equal and the system updates/replace the file each time the download is requested. 
This PR fixes that anomaly and generates reports only for **Finalized** submissions. 

#### How should this be manually tested?
1. Go to the SGA component, and upload **.pdf** file from up to 3 different accounts. 
2. Submit the problem from 2 of 3 accounts and leave only uploaded. 
3. Click the Grade Submissions tab on the SGA xblock
2. All of the written submissions appear in a table. Verify two submissions appear "Submitted: Yes" and for one "Submitted: No"
3. To download all submissions, click the **Download All Submissions** button at the top.
4. It will give a message in red saying: it is preparing the file of all student submissions.
5. When the file is ready, it will give a message in green saying: Student submission file ready for download
6. Click the Download all submission button again.
Expected behavior: Zip file will download
Actual behavior:  System says it is preparing the file of all student submissions.

#### Screenshots (if appropriate)
Screen Shot 2021-02-09 at 7 12 58 PM

Code Review:
@pdpinch

### CODE REVIEW

File: edx_sga/tasks.py [hunk 1]
```diff
@@ -44,7 +50,7 @@ def _get_student_submissions(block_id, course_id, locator):
                 submission['answer']['filename']
             )
         )
-        for submission in submissions if submission['answer']
+        for submission in final_submissions(submissions)
```
Comment (awaisdar001) [inline edx_sga/tasks.py:53]: Loop like a native -- Thanks @nedbat

File: edx_sga/tasks.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 from submissions import api as submissions_api
 
 from edx_sga.constants import ITEM_TYPE
-from edx_sga.utils import get_file_storage_path
+from edx_sga.utils import get_file_storage_path, is_finalized_submission
```
Comment (awaisdar001) [inline edx_sga/tasks.py:16]: @saadyousafarbi @Ali-D-Akbar

**Comment by umarmughal824** [comment]
@awaisdar001 Could you please increase the code coverage by including tests for this PR?

**Comment by umarmughal824** [comment]
@pdpinch I have tested it out, it list down all the files both finalized and non finalized but download only the finalized files.

Screenshot 2021-02-16 at 15 05 01

Screenshot 2021-02-16 at 15 06 45

**Comment by awaisdar001** [comment]
> @awaisdar001 Could you please increase the code coverage by including tests for this PR?

Thanks for taking a look @umarmughal824 Actually this file has been mocked to be used inside tests because it imports an external `from common.djangoapps.student.models import user_by_anonymous_id` 


I tried to mock the call in multiple ways, but I haven't been successful either with `pytest` or `tox`. I think we can make an exception as this is just a minor fix and can be tested manually to skip this test at the moment.

**Review by umarmughal824 — APPROVED** [review]
👍

**Pull request merged**